src/hoi4_world/states/hoi4_states_converter.cpp

```diff
@@ -179,22 +180,36 @@ std::vector<std::set<int>> ConsolidateProvinceSets(std::vector<std::set<int>> co
 }
 
 
-std::vector<hoi4::State> CreateStates(const std::map<int, std::set<int>>& state_id_to_hoi4_provinces,
+std::vector<hoi4::State> CreateStates(const std::map<int, vic3::State>& states,
+    const std::map<int, std::set<int>>& state_id_to_hoi4_provinces,
     const maps::MapData& map_data,
     const maps::ProvinceDefinitions& hoi4_province_definitions,
-    const hoi4::StrategicRegions& strategic_regions)
+    const hoi4::StrategicRegions& strategic_regions,
+    const mappers::CountryMapper& country_mapper)
 {
    std::vector<hoi4::State> hoi4_states;
    for (const auto& [state_id, hoi4_provinces]: state_id_to_hoi4_provinces)
    {
+      const auto state_itr = states.find(state_id);
+      if (state_itr == states.end())
+      {
+         // I can't think how this would happen, so we'd better force people to the forums if it does.
+         throw std::runtime_error("Something has gone very wrong.");
+      }
+      std::optional<std::string> state_owner;
+      if (const std::optional<std::string>& source_owner = state_itr->second.GetOwnerTag(); source_owner.has_value())
+      {
+         state_owner = country_mapper.GetHoiTag(*source_owner);
```

Comment:
This is the bit where I'd preemptively put a warning check instead. I know you return a mock, but you should IMHO return a std::optional mock, and handle the consequences of a broken mapping on state ownership (potentially user error if they massacre the mapping file).

Reply:
Done.